diff --git a/src/Controller/ImageStyleDownloadController.php b/src/Controller/ImageStyleDownloadController.php
index c1718be..44cc10d 100644
--- a/src/Controller/ImageStyleDownloadController.php
+++ b/src/Controller/ImageStyleDownloadController.php
@@ -128,9 +128,16 @@ class ImageStyleDownloadController extends FileDownloadController {
         $possible_image_uris[] = str_replace('.webp', $extension, $image_uri);
       }
 
+      // Search source image.
       foreach ($possible_image_uris as $possible_image_uri) {
-        if (file_exists($possible_image_uri)) {
-          $image_uri = $possible_image_uri;
+        /** @var \Drupal\file\FileInterface[] $files */
+        $files = \Drupal::entityTypeManager()
+                        ->getStorage('file')
+                        ->loadByProperties(['uri' => $possible_image_uri]);
+        /** @var \Drupal\file\FileInterface|null $file */
+        $file = reset($files) ?: NULL;
+        if ($file) {
+          $image_uri = $file->getFileUri();
           break;
         }
       }
